pkg/kyc/component.go

```diff
@@ -63,6 +63,11 @@ type AccreditationsServiceClient interface {
 	NotifyCheck(ctx context.Context, check accreditationsclient.CheckRepresentation) error
 }
 
+// UserProfileCache interface
+type UserProfileCache interface {
+	GetRealmUserProfile(ctx context.Context, realmName string) (kc.UserProfileRepresentation, error)
+}
+
 // Component is the register component interface.
 type Component interface {
```

Comment:
If we go with the idea of having apiNames instead of frontendTypes, it woudl make sense to have a user profile endpoint also for kyc API to be coherent and having full flexibility.